src/database/DatabaseInteractions.php

```diff
@@ -24,25 +24,39 @@ function getReadOnlyDatabase() {
 }
 
 function getSingleBlock(string $username, IDatabase $dbr) {
-	$row = $dbr->selectRow('scratch_accountrequest_block', array('block_username', 'block_reason'), ['LOWER(CONVERT(block_username using utf8))' => strtolower($username)], __METHOD__);
+	$row = $dbr->selectRow('scratch_accountrequest_block', array('block_username', 'block_reason', 'block_expiration_timestamp'), ['LOWER(CONVERT(block_username using utf8))' => strtolower($username)], __METHOD__);
 	return $row ? AccountRequestUsernameBlock::fromRow($row) : false;
 }
 
-function addBlock(string $username, string $reason, User $blocker, IDatabase $dbw) {
+function addBlock(string $username, string $reason, ?string $expirationTimestamp, User $blocker, IDatabase $dbw) {
```

Comment:
This should probably use an upsert instead of an insert in case there are any non-purged expired blocks of the username.

Reply:
From what I know this is not a problem; `handleBlockFormSubmission` upserts. Note that getSingleBlock still returns expired blocks.

```php
$block = getSingleBlock($username, $dbw);
if ($block) {
	updateBlock($username, $reason, $expirationTimestamp, $this->getUser(), $dbw);
} else {
	if ($expirationTimestamp === 'existing') $expirationTimestamp = null;
	addBlock($username, $reason, $expirationTimestamp, $this->getUser(), $dbw);
}
```